File: 1762. Buildings With an Ocean View/forward.monotonicStack.js
/**
 * @param {number[]} heights
 * @return {number[]}
 */
 var findBuildings = function(heights) {
    const n = heights.length;
    const stack = [];
    
    const top = () => stack[stack.length-1]
    
    for(let i = 0; i < n; i++){
        while(stack.length && heights[top()] <= heights[i]){
            stack.pop();
        }
        
        stack.push(i);
    }
    
    return stack;
};

File: 1762. Buildings With an Ocean View/forward.monotonicStack.py
class Solution:
    def findBuildings(self, heights: List[int]) -> List[int]:
        n = len(heights);
        stack = [];
        
        for i in range(n):
            while len(stack) > 0 and heights[stack[-1]] <= heights[i]:
                stack.pop();
            
            stack.append(i)
        
        return stack;
        